Source/Diagnostics/ReducedDiags/FieldReduced.cpp

```diff
@@ -0,0 +1,258 @@
+        const MultiFab & Ex = warpx.getEfield(lev,0);
+        const MultiFab & Ey = warpx.getEfield(lev,1);
+        const MultiFab & Ez = warpx.getEfield(lev,2);
+        const MultiFab & Bx = warpx.getBfield(lev,0);
+        const MultiFab & By = warpx.getBfield(lev,1);
+        const MultiFab & Bz = warpx.getBfield(lev,2);
+
+        if (m_fieldEnergy)
+        {
+            // get cell size
+            Geometry const & geom = warpx.Geom(lev);
+            auto domain_box = geom.Domain();
+#if (AMREX_SPACEDIM == 2)
+            auto dV = geom.CellSize(0) * geom.CellSize(1);
+#elif (AMREX_SPACEDIM == 3)
+            auto dV = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2);
+#endif
+
+            // compute E squared
+            Real tmpx = Ex.norm2(0,geom.periodicity());
+            Real tmpy = Ey.norm2(0,geom.periodicity());
+            Real tmpz = Ez.norm2(0,geom.periodicity());
+            Real Es = tmpx*tmpx + tmpy*tmpy + tmpz*tmpz;
+
+            // compute B squared
+            tmpx = Bx.norm2(0,geom.periodicity());
+            tmpy = By.norm2(0,geom.periodicity());
+            tmpz = Bz.norm2(0,geom.periodicity());
+            Real Bs = tmpx*tmpx + tmpy*tmpy + tmpz*tmpz;
+
+            // save data
+            m_data[lev*3+1] = 0.5 * Es * PhysConst::ep0 * dV;
+            m_data[lev*3+2] = 0.5 * Bs / PhysConst::mu0 * dV;
+            m_data[lev*3+0] = m_data[lev*3+1] + m_data[lev*3+2];
+        }
+
+        if (m_maxField)
+        {
+            // get Maximums of E field components
+            m_data[m_offset_maxField+lev*8] = Ex.norm0();
+            m_data[m_offset_maxField+lev*8+1] = Ey.norm0();
+            m_data[m_offset_maxField+lev*8+2] = Ez.norm0();
+
+            // get Maximums of B field components
+            m_data[m_offset_maxField+lev*8+4] = Bx.norm0();
+            m_data[m_offset_maxField+lev*8+5] = By.norm0();
+            m_data[m_offset_maxField+lev*8+6] = Bz.norm0();
+
+            // Create temporary MultiFAB to be filled with |E| and |B| squared
+            // Note that allocating a full MultiFAB is probably not optimal for memory usage
+            const int ncomp = 1;
+            const int ngrow = 0;  // no ghost cells for temporary MultiFAB
+            MultiFab mftemp(amrex::convert(Ex.boxArray(), IntVect{AMREX_D_DECL(0,0,0)}),
+                                                Ex.DistributionMap(), ncomp, ngrow);
+            // Temporary MultiFab is cell-centered so that it can be filled for any staggering
+            // (possibly with the sum of components which do not have the same staggering).
+
+            MFItInfo info;
+            if (TilingIfNotGPU()) {
+                info.EnableTiling();
+            }
+#ifdef _OPENMP
+            info.SetDynamic(WarpX::do_dynamic_scheduling);
+#pragma omp parallel
+#endif
+            // MFIter loop to fill temporary MultiFAB with |E| squared.
+            for (MFIter mfi(mftemp, info); mfi.isValid(); ++mfi )
```

Comment:
This looks like it's probably outdated syntax.
We usually do compact constructs like this these days:
https://github.com/ECP-WarpX/WarpX/blob/8d73c267d5ed28fc7bd8ef26b7360d667d8ea2ed/Source/FieldSolver/FiniteDifferenceSolver/EvolveF.cpp#L72-L75

Maybe we can update the place where you found this, too?

Reply:
Thanks for the info! I had used the same construct in the `doQEDSchwinger` function so I've updated it there as well.

Initially I found it here:

https://github.com/ECP-WarpX/WarpX/blob/8d73c267d5ed28fc7bd8ef26b7360d667d8ea2ed/Source/Particles/PhysicalParticleContainer.cpp#L577-L585

But I don't know if I should update it there as well; maybe this construct is necessary if we explicitly want dynamic scheduling?

Reply:
Dynamic scheduling: I think we control this centrally via `warpx.do_dynamic_scheduling` these days. Would not adjust in the current location.